### ISSUE TITLE
segfaults on v1.5.0-beta1

### ISSUE THREAD

**Issue by yakir12** [JuliaPlots/Makie.jl#627]
(This has been discussed on slack#makie-cairo)

Pretty simple plotting attempts end in segmentation faults in Julia `v1.5.0-beta1` mentioning Cairo. 

To try this out, 
```julia
] add https://github.com/yakir12/Fake.jl
```
All the `bad` functions cause segfaults.
```julia
using Fake
Fake.bad1() # segfault
Fake.bad2() # segfault
Fake.bad3() # segfault
Fake.bad4() # segfault
```
All the `good` ones don't.
```julia
Fake.good1()
Fake.good2()
Fake.good3()
```
See the [code](https://github.com/yakir12/Fake.jl/blob/master/src/Fake.jl) for the differences between the functions, but it's basically with/out text, with/out `linesegments`, and with/out any plotting.

Here's the stack-trace:
```
yakir@eira:~/tmp/tmp/tmp$ julia --project=. -e "using Fake; Fake.bad1()"

signal (11): Segmentation fault
in expression starting at none:1
unsafe_convert at ./pointer.jl:65 [inlined]
unsafe_convert at ./pointer.jl:66 [inlined]
iswritable at ./iostream.jl:73 [inlined]
unsafe_write at ./iostream.jl:372
unsafe_write at ./io.jl:342 [inlined]
unsafe_write at ./io.jl:627 [inlined]
write_to_stream_callback at /home/yakir/.julia/packages/Cairo/Xr8E0/src/Cairo.jl:158
unknown function (ip: 0x7f9b4e369be4)
_cairo_output_stream_write at /workspace/srcdir/cairo-1.16.0/src/cairo-output-stream.c:267
_cairo_output_stream_write at /workspace/srcdir/cairo-1.16.0/src/cairo-output-stream.c:264 [inlined]
_cairo_output_stream_vprintf at /workspace/srcdir/cairo-1.16.0/src/cairo-output-stream.c:456
_cairo_output_stream_printf at /workspace/srcdir/cairo-1.16.0/src/cairo-output-stream.c:529
_cairo_pdf_surface_start_page at /workspace/srcdir/cairo-1.16.0/src/cairo-pdf-surface.c:2348
_start_page at /workspace/srcdir/cairo-1.16.0/src/cairo-paginated-surface.c:542
_cairo_paginated_surface_show_page at /workspace/srcdir/cairo-1.16.0/src/cairo-paginated-surface.c:579
_cairo_paginated_surface_finish at /workspace/srcdir/cairo-1.16.0/src/cairo-paginated-surface.c:205
_cairo_surface_finish at /workspace/srcdir/cairo-1.16.0/src/cairo-surface.c:1030
cairo_surface_finish at /workspace/srcdir/cairo-1.16.0/src/cairo-surface.c:1079
finish at /home/yakir/.julia/packages/Cairo/Xr8E0/src/Cairo.jl:258
backend_show at /home/yakir/.julia/packages/CairoMakie/vD23B/src/infrastructure.jl:260
show at /home/yakir/.julia/packages/AbstractPlotting/q9DyS/src/display.jl:94
unknown function (ip: 0x7f9b4e368a65)
#873 at /home/yakir/.julia/packages/AbstractPlotting/q9DyS/src/display.jl:272
#open#291 at ./io.jl:330
open at ./io.jl:328
#save#872 at /home/yakir/.julia/packages/AbstractPlotting/q9DyS/src/display.jl:271
save at /home/yakir/.julia/packages/AbstractPlotting/q9DyS/src/display.jl:256
unknown function (ip: 0x7f9b4e364f75)
#save#15 at /home/yakir/.julia/packages/FileIO/ZknoK/src/loadsave.jl:118
save at /home/yakir/.julia/packages/FileIO/ZknoK/src/loadsave.jl:118 [inlined]
bad1 at /home/yakir/.julia/packages/Fake/haPLE/src/Fake.jl:11
unknown function (ip: 0x7f9b9828d6ac)
jl_apply at /home/yakir/julia/src/julia.h:1701 [inlined]
do_call at /home/yakir/julia/src/interpreter.c:117
eval_value at /home/yakir/julia/src/interpreter.c:206
eval_stmt_value at /home/yakir/julia/src/interpreter.c:157 [inlined]
eval_body at /home/yakir/julia/src/interpreter.c:548
jl_interpret_toplevel_thunk at /home/yakir/julia/src/interpreter.c:660
jl_toplevel_eval_flex at /home/yakir/julia/src/toplevel.c:840
jl_toplevel_eval_flex at /home/yakir/julia/src/toplevel.c:790
jl_toplevel_eval_flex at /home/yakir/julia/src/toplevel.c:790
jl_toplevel_eval_in at /home/yakir/julia/src/toplevel.c:883
eval at ./boot.jl:331
exec_options at ./client.jl:260
_start at ./client.jl:485
jfptr__start_32753 at /home/yakir/julia/usr/lib/julia/sys.so (unknown line)
unknown function (ip: 0x555d364fc702)
unknown function (ip: 0x555d364fc322)
__libc_start_main at /lib/x86_64-linux-gnu/libc.so.6 (unknown line)
unknown function (ip: 0x555d364fc3c9)
Allocations: 102987213 (Pool: 102966811; Big: 20402); GC: 84
Segmentation fault
```

**Comment by SimonDanisch** [comment]
90% Sure, that this should be a cairo issue!
The function in which this segfaults looks super dubious:
https://github.com/JuliaGraphics/Cairo.jl/blob/master/src/Cairo.jl#L157

**Comment by asinghvi17** [comment]
What does this even mean?  🤣 
https://github.com/JuliaGraphics/Cairo.jl/blob/25f5658c9b70075bd3f1a781febc5a38c5cbd002/src/Cairo.jl#L158

**Comment by jkrumbiegel** [comment]
fixed with new CairoMakie version 0.2.9, cf. https://github.com/JuliaPlots/CairoMakie.jl/commit/bea0c0d3b8ef6b628148b9eb316ade862ba96b0a
